var/spack/repos/builtin/packages/py-meson-python/package.py

```diff
@@ -21,3 +23,6 @@ class PyMesonPython(PythonPackage):
     depends_on("py-tomli@1:", type=("build", "run"))
     depends_on("py-typing-extensions@3.7.4:", when="^python@:3.7", type=("build", "run"))
     depends_on("py-colorama", when="platform=windows", type=("build", "run"))
+
+    # https://github.com/FFY00/meson-python/pull/111
+    # conflicts('platform=darwin os=bigsur:', when='@:0.7')
```

Comment:
@becker33 @alalazo this doesn't work, how do I make it work? I want to say that version 0.7 and older doesn't work on macOS 11+ (big sur, monterey, ventura, etc.)

Reply:
I think the part that doesn't work is `os=bigsur:` with the colon. We don't have hierarchies for operating systems, in fact they're modeled as strings in a spec:
```
>>> s = spack.spec.Spec('py-scipy os=bigsur:')
>>> s.architecture.os
'bigsur:'
```
Can you check if listing a few operating systems explicitly works?

Reply:
Yes, the following works:
```python
    conflicts('platform=darwin os=ventura', when='@:0.7')                                           
    conflicts('platform=darwin os=monterey', when='@:0.7')                                          
    conflicts('platform=darwin os=bigsur', when='@:0.7')          
```
but would obviously need to be updated for every new OS. I'm quite surprised that we've managed to find a way to model architectures with a partial ordering but still haven't given any ordering to OSes.

Is the above (which needs to be changed for every new OS) better than the alternative:
```python
@run_before('install')
def check_os(self):
    if self.spec.satisfies('@:0.7 platform=darwin') and macos_version() >= Version('10'):
        raise SpecError(...)
```
? Not sure which to use until we start ordering OSes.